### DISCUSSION TITLE
Configuration error on ebusd add on in home assistant

### DISCUSSION THREAD

**Discussion by eerke2021** [john30/ebusd#1169] [Q&A]
Under options you can choose enh, ens or udp
It then says: "network adapter address ie. enh:IP:PORT"

My adapter is fixed at 192.168.0.220

If I put enh:192.168.0.220:9999
In the logs I get the error:
"[main error] unable to create device enh:enh:192.168.0.220:9999"

But if I ignore the instructions and put just "192.168.0.220:9999

The log finishes by saying "device status: extra info: firmware 1.1[4106].1[4106], jumpers 0x0f

Which sounds a lot better. Does that mean ebusd is connected to the ebus adapter even though I have ignored the instruction?

**Comment by Koky05** [comment]
Manual was written for older devices. You have adapter v5 and should use `ens:192.168.0.220:9999`

**Reply by eerke2021** [reply]
If I do that I get
"[main error] unable to create device ens:ens:192.168.0.220:9999"
Will now try leaving the ens selection but deleting it from string

**Reply by eerke2021** [reply]
and again last line "2024-01-26 07:01:22.736 [bus notice] device status: extra info: firmware 1.1[4106].1[4106], jumpers 0x0f"
NB boiler not wired in yet. Got ready but not connected. Next step.

**Comment by Koky05** [comment]
Do you use full spec? `network_device: ens:192.168.0.220:9999`, or use mode ens and add just address?
 And yes if you are not connected already to boiler it is correct behavior.

**Reply by eerke2021** [reply]
I'm not sure the full spec applies to the Home Assistant configuration of EBUSD. First it asks the mode with three options enh, and and tcp I think I remember
Then it asks for the network address but there is no underscore between the words and suggests adding the option again but always errors if you follow the instructions. 
I selected ENS of the three options. Then I only put 192.168.0.220:9999
(That is the static IP of my adapter). 
If that looks okay next step I connect boiler and explore MQTT messages.

**Comment by Koky05** [comment]
OK, if you choose mode then your address doesn't require ens or enh as prefix.

This is from config.yaml code of that addon:
```
schema:
  mode: "list(enh|ens|udp)?"
```

But you should choose `ens` as mode.

**Reply by eerke2021** [reply]
Thanks. I have set it to ens and will attach the boiler tonight. I get the impression that 3 mode setting did not used to exist in the Home Assistant Add On configuration but when it was added the instructions for the next bit didn't get updated and so confused me. I did think "why do I need to tell them twice", but then thought I'd just follow the instructions exactly.

**Reply by Koky05** [reply]
It has two way of configuration, via UI or configuration file. In UI you define mode. In configuration file you use code `network_device: ens:192.168.0.220:9999` and that makes it confusing.

**Reply by eerke2021** [reply]
Now the boiler is connected I get extra:
"
2024-01-26 18:41:41.195 [bus notice] new master 03, master count 1
2024-01-26 18:41:49.254 [main error] scan config 08: ERR: element not found
"
I check the wiring

(Wiring checked and all okay)

**Reply by eerke2021** [reply]
mqtt explorer only reveals under ebus/global
ebusd
global
version = "ebusd 23.2.p20230716"
running = true
updatecheck = "version 23.3 available"
scan = "finished"
signal = true
uptime = 272
Seems ebusd is running, MQTT and adapter connected. Just boiler is not (but different than entirely unconnected), but I restarted everything and increased log level to debug. It is a Vaillant ecoTEC Plus 630. I thought this was okay already. I double check now

**Reply by eerke2021** [reply]
I turned off the read only configuration (only optional choice I made other than logging level) and now it connects. There are still occasional errors in the logs but most things are okay. Did I maybe misunderstand what "read only" meant?

Occasional errors in logs are
:"2024-01-26 20:32:50.281 [update error] unable to parse poll-read bai externalFlowTempDesired from 3108b509030d2500 / 00: ERR: invalid position" They mostly are that value or the minimum/maximum value of it. The boiler is off but the desired temperature (if it was ON) is 65°C.
